src/tester.rs

```diff
@@ -651,8 +652,14 @@ fn run_cmd(
         .spawn()
         .unwrap_or_else(|_| fatal(&format!("Couldn't run command {:?}.", cmd)));
 
-    let mut stderr = FileDescriptor::dup(child.stderr.as_ref().unwrap()).unwrap();
     let mut stdout = FileDescriptor::dup(child.stdout.as_ref().unwrap()).unwrap();
+    let mut stderr = FileDescriptor::dup(child.stderr.as_ref().unwrap()).unwrap();
+
+    fcntl(stdout.as_socket_descriptor(), F_SETFL(OFlag::O_NONBLOCK))
```

Comment:
I'm pretty sure that you need to do `flags = fcntl(..., F_GETFL, 0)` to read the flags, then `fcntl(..., F_SETFL, flags | O_NONBLOCK)`.